instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/SnippetInjectingResponseWrapper.java

```diff
@@ -144,8 +152,16 @@ public PrintWriter getWriter() throws IOException {
     }
     if (snippetInjectingPrintWriter == null) {
       snippetInjectingPrintWriter =
-          new SnippetInjectingPrintWriter(super.getWriter(), SNIPPET, super.getCharacterEncoding());
+          new SnippetInjectingPrintWriter(super.getWriter(), SNIPPET, this);
     }
     return snippetInjectingPrintWriter;
   }
+
+  public void updateContentLength() {
+    setContentLength((int) contentLength + SNIPPET_LENGTH);
+  }
+
+  public boolean contentLengthWasSet() {
+    return contentLength != -1;
+  }
```

Comment:
since these are always called together, it may be nice to combine them into a single call, e.g. `updateContentLengthIfPreviouslySet()`